### PR TITLE
Add vacuum animation

### PR DESCRIPTION
**Pull request by marcgeurts**
## Description
Added optional vacuum and docking animation to the vacuum card.

## Related Issue

[#846](https://github.com/piitaya/lovelace-mushroom/issues/846)

## Types of changes

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [x] 🌎 Translation (addition or update a translation)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have tested the change locally.
- [x] I followed [the steps](https://github.com/piitaya/lovelace-mushroom#maintainer-steps-to-add-a-new-language
) if I add a new language .

### CODE REVIEW

**Review by piitaya — CHANGES_REQUESTED** [review]
Thanks for the PR. Your `package-lock` is different because you use a older version of npm < 6. I think you should use npm > 7 or avoid commiting the `package-lock`.

**Comment by marcgeurts** [comment]
> Thanks for the PR. Your `package-lock` is different because you use a older version of npm < 6. I think you should use npm > 7 or avoid commiting the `package-lock`.

I removed the package-lock

File: src/cards/vacuum-card/vacuum-card.ts [hunk 1]
```diff
@@ -135,6 +168,12 @@ export class VacuumCard extends MushroomBaseCard implements LovelaceCard {
                     --icon-color: rgb(var(--rgb-state-vacuum));
                     --shape-color: rgba(var(--rgb-state-vacuum), 0.2);
                 }
+                mushroom-shape-icon.cleaning {
+                    --icon-animation: var(--animation-duration) infinite linear cleaning;
+                }
+                mushroom-shape-icon.returning {
+                    --icon-animation: var(--animation-duration) infinite linear returning;
+                }
```
Comment (piitaya) [inline src/cards/vacuum-card/vacuum-card.ts:176]:
```suggestion
                mushroom-shape-icon.cleaning {
                    --icon-animation: 5s infinite linear cleaning;
                }
                mushroom-shape-icon.returning {
                    --icon-animation: 2s infinite linear returning;
                }
```

File: src/cards/vacuum-card/vacuum-card.ts [hunk 2]
```diff
@@ -93,36 +97,65 @@ export class VacuumCard extends MushroomBaseCard implements LovelaceCard {
...
+        const cleaning = isCleaning(entity);
+        if (active) {
+            if (cleaning) {
+                iconStyle["--animation-duration"] = `5s`;
+            }
+        } else {
+            if (returning) {
+                iconStyle["--animation-duration"] = `2s`;
+            }
+        }
```
Comment (piitaya) [inline src/cards/vacuum-card/vacuum-card.ts:143]:
```suggestion
```
No needed

**Review by piitaya — CHANGES_REQUESTED** [review]

File: src/cards/vacuum-card/vacuum-card.ts [hunk 3]
```diff
@@ -93,36 +97,53 @@ export class VacuumCard extends MushroomBaseCard implements LovelaceCard {
...
+
+        return html`
+            <mushroom-shape-icon
+                slot="icon"
+                class=${classMap({
+                    returning: isReturningHome(entity) && Boolean(this._config?.icon_animation),
+                    cleaning: isCleaning(entity) && Boolean(this._config?.icon_animation),
+                })}
+                style=${styleMap(iconStyle)}
+                .disabled=${!isActive(entity) && !isReturningHome(entity)}
```
Comment (piitaya) [inline src/cards/vacuum-card/vacuum-card.ts:141]:
```suggestion
                .disabled=${!isActive(entity)}
```
Should be enough as returning home is now an active state with https://github.com/piitaya/lovelace-mushroom/pull/855

Comment (marcgeurts) [reply src/cards/vacuum-card/vacuum-card.ts]: Nice!

**Pull request merged**